### DISCUSSION TITLE
Cross Compiling Tools on Fedora Containers

### DISCUSSION THREAD

**Discussion by cfernald** [tianocore/containers#71] [General]
Hello @osteffenrh, I wanted to get your opinion on an issue that I ran into yesterday regarding the Fedora container. This may be specific to Project MU's use case, but I think has some long term affects. 

We pre-compile basetools and provide them as an external dependency to speed up the pipeline and build process. To do this we need to cross compile for the different support dev environments, e.g. AARCH64. From the package documentation it seems that this is not a supported use-case currently in the fedora packages which only support cross compiling kernels.

While this blocks our use case of the Fedora container this is not something I believe EDKII generally requires. I'd be curious to see if you see this as an issue or not or if anything could be done about it.

**Comment by osteffenrh** [comment]
Hi Chris,

Quoting Chris Fernald (2023-05-02 19:39:42)
> Hello ***@***.***, I wanted to get your opinion on an issue that I ran into
> yesterday regarding the Fedora container. This may be specific to Project MU's
> use case, but I think has some long term affects.
>
> We pre-compile basetools and provide them as an external dependency to speed up
> the pipeline and build process. To do this we need to cross compile for the
> different support dev environments, e.g. AARCH64. From the package
> documentation it seems that this is not a supported use-case currently in the
> fedora packages which only support cross compiling kernels.
That seems to be the case, unfortunately.

Are you intending using the AARCH64 BaseTools artifacts in the CI on
AARCH64? If so, could you build BaseTools also on the same arch? We
could add a docker image for aarch64, then you would not need to
cross-compile.

**Reply by ghost** [reply]
Since I have several AArch64 systems, I created https://github.com/tianocore/containers/pull/92 to start adding support for building arm64 containers alongside amd64.
